test/pylib/scylla_server.py

```diff
@@ -215,9 +216,18 @@ async def cql_is_up(self) -> bool:
         try:
             with Cluster(execution_profiles={EXEC_PROFILE_DEFAULT: profile},
                          contact_points=[self.hostname], auth_provider=auth) as cluster:
-                with cluster.connect():
+                with cluster.connect() as session:
+                    # auth::standard_role_manager creates "cassandra" role in an
+                    # async loop auth::do_after_system_ready(), which retries
+                    # role creation with an exponential back-off. In other
+                    # words, even after CQL port is up, Scylla may still be
+                    # initializing. When the role is ready, queries begin to
+                    # work, so rely on this "side effect".
+                    session.execute("CREATE KEYSPACE k WITH REPLICATION = {" +
+                                    "'class' : 'SimpleStrategy', 'replication_factor' : 1 }")
+                    session.execute("DROP KEYSPACE k")
```

Comment:
Is the keyspace creation/deletion somehow needed for testing the "cassandra" role?
In my code, I used the silly request "use system" as a do-nothing request. But maybe that doesn't fail without the role?

Anyway, it seems to me that you actually discovered a Scylla bug, and we shouldn't paper over it with a test fix (although we can - temporarily).

Reply:
If you look at the test error log, force_schema_migration() fails on partially initialized instances. force_schema_migration() uses create/drop keyspace. This is why I use the same statements. I haven't checked for a minimal statement that would trigger the problem. use is only calling validate_login() so not necessarily checking the role.